tests/test_magpie_api.py

```diff
@@ -107,6 +111,155 @@ def setUpClass(cls):
+        """
+        # Create temporary config for testing webhooks
+        with tempfile.NamedTemporaryFile(mode="wt") as tmp_config:
+            base_webhook_url = get_constant("MAGPIE_TEST_USER_WEBHOOK_URL")
+            delete_webhook_url = base_webhook_url + "/webhook"
+            data = {
+                "webhooks":
+                    # Use two identical urls to simulate having multiple webhook urls
+                    {"delete": [delete_webhook_url, delete_webhook_url]}}
+            yaml.dump(data, tmp_config, default_flow_style=False)
+
+            with mock.patch.dict(os.environ, {"MAGPIE_CONFIG_PATH": tmp_config.name}):
+                utils.get_test_webhook_app()
+                # create the test user first
+                utils.TestSetup.create_TestUser(self, override_group_name=get_constant("MAGPIE_ANONYMOUS_GROUP"))
+
+                # Webhooks shouldn't have been called during the user creation
+                sleep(1)
+                resp = requests.get(base_webhook_url + "/get_status")
+                assert resp.text == "0"
+
+                # delete the test user, webhooks should be called during this delete request
+                path = "/users/{usr}".format(usr=self.test_user_name)
+                resp = utils.test_request(self, "DELETE", path, headers=self.json_headers, cookies=self.cookies)
+                utils.check_response_basic_info(resp, 200, expected_method="GET")
+                utils.TestSetup.check_NonExistingTestUser(self)
+
+                # Wait for the webhook requests to complete and check their success
+                sleep(1)
+                resp = requests.get(base_webhook_url + "/get_status")
+                assert resp.text == "2"
+
+    def test_Webhook_DeleteUser_FailingUrl(self):
+        """
+        Test deleting a user using a failing webhook url.
+        """
+        # Create temporary config for testing webhooks
+        with tempfile.NamedTemporaryFile(mode="wt") as tmp_config:
+            delete_webhook_url = "failing_url"
+            data = {"webhooks": {"delete": [delete_webhook_url]}}
+            yaml.dump(data, tmp_config, default_flow_style=False)
+
+            with mock.patch.dict(os.environ, {"MAGPIE_CONFIG_PATH": tmp_config.name}):
+                # create the test user first
+                utils.TestSetup.create_TestUser(self, override_group_name=get_constant("MAGPIE_ANONYMOUS_GROUP"))
+
+                # delete the test user, webhooks should be called during the request
+                path = "/users/{usr}".format(usr=self.test_user_name)
+                resp = utils.test_request(self, "DELETE", path, headers=self.json_headers, cookies=self.cookies)
+
+                # Check if user deletion was successful even if the webhook failed
+                sleep(1)
+                utils.check_response_basic_info(resp, 200, expected_method="GET")
+                utils.TestSetup.check_NonExistingTestUser(self)
```

Comment:
There is nothing that seems to validate that Magpie sent the request to the webhook URL. 
Should have some kind of mock of the error callback to ensure it was called.